include/libopencm3/sam/4l/gpio.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2016 Maxim Sloyko <maxim@sloyko.com>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBOPENCM3_GPIO_H
+#define LIBOPENCM3_GPIO_H
+
+#include <libopencm3/cm3/common.h>
+#include <libopencm3/sam/memorymap.h>
+#include <libopencm3/sam/common/gpio_common_all.h>
+
+#define GPIOA	(GPIO_BASE)
+#define GPIOB	(GPIO_BASE + 0x200)
+#define GPIOC	(GPIO_BASE + 0x400)
+
+
+
+#define GPIO_GPER(X)			MMIO32((X) + 0x10 * 0)
+#define GPIO_GPERS(X)			MMIO32((X) + 0x10 * 0 + 0x4)
+#define GPIO_GPERC(X)			MMIO32((X) + 0x10 * 0 + 0x8)
+#define GPIO_GPERT(X)			MMIO32((X) + 0x10 * 0 + 0xc)
```

Comment:
Are these set/clear/toggle registers?  sounds cute :)  as an idea, would it be nicer from the user side to be GPIO_SET(GPIO_GPER(GPIOA), xxx) and GPIO_TOGGLE(GPIO_PMR0(GPIOB), xxx) instead of having extra definitions?  probably doesn't matter much either way, just thinking out loud over how the user would like use these functions